src/main/java/com/commercetools/project/sync/util/ClientConfigurationUtils.java

```diff
@@ -29,25 +19,23 @@
   private static final long DEFAULT_TIMEOUT = 30000;
 
   /**
-   * Creates a {@link BlockingSphereClient} with a custom {@code timeout} with a custom {@link
-   * TimeUnit}.
+   * Creates a {@link SphereClient} with a default {@code timeout} value of 30 seconds.
    *
    * @param clientConfig the client configuration for the client.
-   * @return the instantiated {@link BlockingSphereClient}.
+   * @return the instantiated {@link SphereClient}.
    */
   public static SphereClient createClient(@Nonnull final SphereClientConfig clientConfig) {
 
     final HttpClient httpClient = getHttpClient();
-    final SphereAccessTokenSupplier tokenSupplier =
-        SphereAccessTokenSupplier.ofAutoRefresh(clientConfig, httpClient, false);
-    final SphereClient underlying = SphereClient.of(clientConfig, httpClient, tokenSupplier);
-    final SphereClient retryClient = withRetry(underlying);
-    return withLimitedParallelRequests(retryClient);
+    return RetryableSphereClientBuilder.of(clientConfig, httpClient)
+        .withStatusCodesToRetry(
+            Arrays.asList(BAD_GATEWAY_502, SERVICE_UNAVAILABLE_503, GATEWAY_TIMEOUT_504))
+        .build();
   }
 
   /**
-   * Gets an asynchronous {@link HttpClient} to be used by the {@link BlockingSphereClient}. Client
-   * is created during first invocation and then cached.
+   * Gets an asynchronous {@link HttpClient} to be used by the {@link SphereClient}. Client is
+   * created during first invocation and then cached.
    *
    * @return {@link HttpClient}
    */
```

Comment:
I think we could use the same client that we have in the latest java-sync, the defaults in DefaultAsyncHttpClientConfig is good. So DEFAULT_TIMEOUT also not needed.

so 

``` java
/**
     * Gets an asynchronous {@link HttpClient} of `asynchttpclient` library,
     * to be used by as an underlying http client for the {@link SphereClient}.
     *
     * @return an asynchronous {@link HttpClient}
     */
    private static HttpClient getHttpClient() {
        final AsyncHttpClient asyncHttpClient =
            new DefaultAsyncHttpClient(new DefaultAsyncHttpClientConfig.Builder().build());
        return AsyncHttpClientAdapter.of(asyncHttpClient);
    }
```

Reply:
c61c772